pre-docs/running_bolt_from_docker_image.md

```diff
@@ -0,0 +1,182 @@
+
+```yaml
+---
+nodes:
+  - name: pnz2rzpxfzp95hh.delivery.puppetlabs.net
+    alias: docker-example
+    config:
+      transport: ssh
+      ssh:
+        user: root
+        password: secret-password
+        host-key-check: false
+```
+The following invocation shows an example of running the built-in `facts` task against the target listed in inventory. 
+```
+cas@cas-ThinkPad-T460p:~/working_dir/docker_bolt$ docker run --env "BOLT_INVENTORY=$(cat Boltdir/inventory.yaml)" puppet/puppet-bolt task run facts -t docker-example
+Started on pnz2rzpxfzp95hh.delivery.puppetlabs.net...
+Finished on pnz2rzpxfzp95hh.delivery.puppetlabs.net:
+  {
+    "os": {
+      "name": "CentOS",
+      "release": {
+        "full": "7.2",
+        "major": "7",
+        "minor": "2"
+      },
+      "family": "RedHat"
+    }
+  }
+Successful on 1 node: pnz2rzpxfzp95hh.delivery.puppetlabs.net
+Ran on 1 node in 0.55 seconds
+```
+## Mount Bolt project directory from host 
+
+This section describes making a Bolt project directory (Boltdir) available to the container. The directory structure and relevant file content is listed below:
+```
+cas@cas-ThinkPad-T460p:~/working_dir/docker_bolt$ tree
+.
+└── Boltdir
+    ├── bolt.yaml
+    ├── inventory.yaml
+    ├── keys
+    │   └── id_rsa-acceptance
+    └── site-modules
+        └── docker_task
+            └── tasks
+                └── init.sh
+
+5 directories, 4 files
+```
+**bolt.yaml**
+
+Bolt configuration file
+```yaml
+---
+log:
+  console:
+    level: notice
+```
+**inventory.yaml**
+
+Store information about targets. Note the absolute path to the private key is the path in the container not the host.
+```yaml
+---
+nodes:
+  - name: pnz2rzpxfzp95hh.delivery.puppetlabs.net
+    alias: docker-example
+    config:
+      transport: ssh
+      ssh:
+        user: root
+        private-key: /Boltdir/keys/id_rsa-acceptance
+        host-key-check: false
+```
+**init.sh**
+
+Example shell task that echos a `message` parameter.
+```bash
+#!/bin/bash
+echo "Message: ${PT_message}"
+```
+In order to execute the task with docker and provide all the information provided in the Bolt project directory it is possible to mount the Boltdir on the host with the following invocation:
+```
+cas@cas-ThinkPad-T460p:~/working_dir/docker_bolt$ docker run --mount type=bind,source=/home/cas/working_dir/docker_bolt/Boltdir,destination=/Boltdir puppet/puppet-bolt task run docker_task message=hi -t docker-example
+Started on pnz2rzpxfzp95hh.delivery.puppetlabs.net...
+Finished on pnz2rzpxfzp95hh.delivery.puppetlabs.net:
+  Message: hi
+  {
+  }
+Successful on 1 node: pnz2rzpxfzp95hh.delivery.puppetlabs.net
+Ran on 1 node in 0.56 seconds
+```
+
+The `--mount` flag maps the local directory to the root directory in the container (the working directory), the container is tagged as `puppet-bolt` and the rest of the invocation are all native to Bolt. 
```

Comment:
This is a little confusing. It might be clearer to say it maps the local `Boltdir` to `/Boltdir`, which is where the container image is configured to look for it.